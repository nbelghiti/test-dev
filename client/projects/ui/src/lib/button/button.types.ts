export enum ButtonType {
	primary = '',
	secondary = 'c-button--secondary',
	tertiary = 'c-button--tertiary',
	destructive = 'c-button--destructive',
}
