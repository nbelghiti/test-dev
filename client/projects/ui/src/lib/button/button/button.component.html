<button
	*ngIf="!link; else isLink"
	class="c-button {{ type }} {{ classes }}"
	[title]="buttonTitle"
	[ngClass]="{
		'c-button--disabled': disabled,
		'c-button--loading': loading,
		'c-button--fluid': fluid
	}"
	[disabled]="disabled"
	(click)="loading ? null : clicked.emit($event)"
>
	<ng-container [ngTemplateOutlet]="buttonContent"></ng-container>
</button>

<ng-template #isLink>
	<a
		class="c-button {{ type }} {{ classes }}"
		[ngClass]="{
			'c-button--disabled': disabled,
			'c-button--loading': loading,
			'c-button--fluid': fluid
		}"
		[routerLink]="!disabled && link"
	>
		<ng-container [ngTemplateOutlet]="buttonContent"></ng-container>
	</a>
</ng-template>

<ng-template #buttonContent>
	<div
		*ngIf="iconBefore"
		class="c-button__icon c-button__icon--before"
	>
		<ui-icon
			[name]="iconBefore"
			[size]="type === 'c-button--tertiary' && 'small'"
		>
		</ui-icon>
	</div>

	<span class="c-button__text">
		<ng-content></ng-content>
	</span>

	<i class="c-button__loader fas fa-spinner fa-spin"></i>

	<div
		*ngIf="iconAfter"
		class="c-button__icon c-button__icon--after"
	>
		<ui-icon
			[name]="iconAfter"
			[size]="type === 'c-button--tertiary' && 'small'"
		>
		</ui-icon>
	</div>
</ng-template>
