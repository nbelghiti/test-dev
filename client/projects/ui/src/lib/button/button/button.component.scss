// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Button
// =============================================================================
.c-button {
	position: relative;

	display: inline-flex;
	justify-content: center;
	align-items: center;
	min-width: 8rem;

	padding: 1rem 2rem;

	text-align: center;
	text-decoration: none;

	border: $border-width-base solid $primary-color;
	background-color: $primary-color;
	color: $text-color-white;

	transition: background-color $animation-base, border-color $animation-base;

	cursor: pointer;

	&:hover {
		background-color: transparent;
		color: $primary-color;
	}

	&:active {
		background-color: rgba($primary-color, 0.08);

		transition: background-color 0s;
	}
}

// =============================================================================
// :: Elements
// =============================================================================
.c-button__icon {
	display: flex;

	font-size: 1.5rem;
	line-height: $line-height-base;

	color: currentColor;

	transition: color $animation-base;
}

.c-button__text {
	display: inline-block;
	max-width: 25rem;
	overflow: hidden;

	font-family: $font-family-secondary;
	font-weight: $font-weight-bold;
	font-size: $font-size-base;
	line-height: $line-height-base;
	text-transform: uppercase;
	text-overflow: ellipsis;
	white-space: nowrap;

	color: currentColor;

	transition: color $animation-base;
}

.c-button__loader {
	position: absolute;

	display: none;

	transform: translate(-50%, -50%);
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-button--secondary,
.c-button--tertiary {
	border: $border-width-base solid $border-color-grey-76;
	background-color: $background-color-base;
	color: $text-color-grey-46;

	&:active {
		background-color: rgba($background-color-black, 0.08);

		transition: background-color $animation-base;
	}

	&:focus {
		outline-color: $border-color-black;
	}

	&:hover {
		border-color: $border-color-black;
		color: $text-color-base;
	}
}

.c-button--tertiary {
	padding: 0.8rem 1rem;

	.c-button__icon {
		line-height: $line-height-normalize;
	}

	.c-button__text {
		font-size: $font-size-2x-small;
	}

	.c-button__icon--after {
		margin-left: 0.8rem;
	}

	.c-button__icon--before {
		margin-right: 0.8rem;
	}
}

.c-button__icon--after {
	margin-left: 1rem;
}

.c-button__icon--before {
	margin-right: 1rem;
}

.c-button--disabled:not(.c-button--secondary):not(.c-button--tertiary) {
	background-color: $background-color-grey-86;
	border-color: $background-color-grey-86;

	pointer-events: none;
}

.c-button--secondary,
.c-button--tertiary {

	&.c-button--disabled {
		color: $text-color-grey-76;

		pointer-events: none;
	}
}

.c-button--loading {
	pointer-events: none;

	.c-button__text,
	.c-button__icon {
		opacity: 0;
	}

	.c-button__loader {
		display: block;
	}
}

.c-button--fluid {
	width: 100%;

	.c-button__text {
		max-width: 100%;
	}
}

.c-button--destructive {
	color: $ui-color-error-dark;
	background-color: $ui-color-error-light;
	border-color: $ui-color-error-light;

	&:hover,
	&:focus,
	&:active {
		color: $ui-color-error-light;
		background-color: $ui-color-error-dark;
		border-color: $ui-color-error-dark;
	}
}
