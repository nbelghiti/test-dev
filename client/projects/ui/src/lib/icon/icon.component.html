<i class="c-icon__glyph {{ size && 'c-icon__glyph--' + size }} {{ type && 'c-icon__glyph--' + type }} fas fa-{{ name }}"></i>
