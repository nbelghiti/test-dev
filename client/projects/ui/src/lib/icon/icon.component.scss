// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Icon
// =============================================================================
:host {
	display: inline-flex;
	justify-content: center;
	align-items: center;

	font-size: inherit;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-icon__glyph {
	line-height: inherit;

	cursor: pointer;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-icon__glyph--large {
	font-size: 2rem;
}

.c-icon__glyph--small {
	font-size: 1.4rem;
}

.c-icon__glyph--tiny {
	font-size: 1.2rem;
}

.c-icon__glyph--success {
	color: $ui-color-success-dark;
}

.c-icon__glyph--error {
	color: $ui-color-error-dark;
}

.c-icon__glyph--warning {
	color: $ui-color-warning-dark;
}

.c-icon__glyph--info {
	color: $ui-color-info-dark;
}
