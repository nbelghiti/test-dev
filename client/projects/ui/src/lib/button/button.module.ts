import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';

import { IconModule } from '../icon/icon.module';

import { ButtonComponent } from './button/button.component';
import { IconButtonComponent } from './icon-button/icon-button.component';

@NgModule({
	imports: [
		CommonModule,
		IconModule,
		RouterModule,
	],
	declarations: [
		ButtonComponent,
		IconButtonComponent,
	],
	exports: [
		ButtonComponent,
		IconButtonComponent,
	],
})
export class ButtonModule {}
