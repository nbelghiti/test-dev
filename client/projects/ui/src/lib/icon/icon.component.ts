import { Component, HostBinding, Input } from '@angular/core';

import { IconName, IconSize, IconType } from './icon.types';

@Component({
	selector: 'ui-icon',
	templateUrl: './icon.component.html',
	styleUrls: ['./icon.component.scss']
})

export class IconComponent {
	@Input() public name: IconName;
	@Input() public type: IconType = null;
	@Input() public size: IconSize = null;

	@HostBinding('class.c-icon') public className = true;
}
