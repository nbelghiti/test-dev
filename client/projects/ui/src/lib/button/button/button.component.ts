import { Component, EventEmitter, Input, Output } from '@angular/core';

import { ButtonType } from '../button.types';

@Component({
	selector: 'ui-button',
	templateUrl: './button.component.html',
	styleUrls: ['./button.component.scss']
})

export class ButtonComponent {
	@Input() public type: ButtonType;
	@Input() public link?: string;
	@Input() public disabled?: boolean = false;
	@Input() public loading?: boolean = false;
	@Input() public fluid?: boolean = false;
	@Input() public iconBefore?: string;
	@Input() public iconAfter?: string;
	@Input() public classes?: string;
	@Input() public buttonTitle: string = '';

	@Output() public clicked: EventEmitter<MouseEvent> = new EventEmitter<MouseEvent>();
}
