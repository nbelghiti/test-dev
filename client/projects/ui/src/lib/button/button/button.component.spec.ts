import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';

import { IconComponent } from '../../icon/icon.component';

import { ButtonComponent } from './button.component';

describe('ButtonComponent', () => {
	let component: ButtonComponent;
	let fixture: ComponentFixture<ButtonComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			imports: [ RouterTestingModule ],
			declarations: [ ButtonComponent, IconComponent ]
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(ButtonComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});
});
