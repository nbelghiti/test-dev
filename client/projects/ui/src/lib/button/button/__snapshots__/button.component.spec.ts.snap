// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonComponent Should render correctly 1`] = `
<ui-button
  buttonTitle=""
  clicked={[Function EventEmitter]}
  disabled="false"
  fluid="false"
  loading="false"
>
  <button
    class="c-button"
    ng-reflect-klass="c-button  "
    ng-reflect-ng-class="[object Object]"
    title=""
  >
    
    
    <span
      class="c-button__text"
    />
    <i
      class="c-button__loader fas fa-spinner fa-spin"
    />
    
    
  </button>
</ui-button>
`;
